src/encoded/static/components/gene.js

```diff
@@ -0,0 +1,150 @@
+            geneLink = dbxrefHref('HGNC', context.gene_name);
+        } else if (context.organism.name === 'mouse') {
+            const mgiRef = _(context.dbxref).find(ref => ref.substr(0, 4) === 'MGI:');
+            if (mgiRef) {
+                geneLink = dbxrefHref('MGI', mgiRef);
+            }
+        } else if (context.organism.name === 'dmelanogaster' || context.organism.name === 'celegans') {
+            const organismPrefix = context.organism.name === 'dmelanogaster' ? 'FBgn' : 'WBGene';
+            const baseUrl = context.organism.name === 'dmelanogaster' ? globals.dbxrefPrefixMap.FlyBase : globals.dbxrefPrefixMap.WormBase;
+            geneRef = _.find(context.dbxref, ref => ref.indexOf(organismPrefix) !== -1);
+            if (geneRef) {
+                sep = geneRef.indexOf(':') + 1;
+                baseName = geneRef.substring(sep, geneRef.length);
+                geneLink = baseUrl + baseName;
+            }
+        }
+
+        // Set up breadcrumbs
+        const crumbs = [
+            { id: 'Genes' },
+            {
+                id: <i>{context.organism.scientific_name}</i>,
+                query: `organism.scientific_name=${context.organism.scientific_name}`,
+                tip: `${context.organism.scientific_name}`,
+            },
+        ];
+
+        return (
+            <div className={globals.itemClass(context, 'view-item')}>
+                <header className="row">
+                    <div className="col-sm-12">
+                        <Breadcrumbs root="/search/?type=gene" crumbs={crumbs} />
+                        <h2>{context.symbol} (<em>{context.organism.scientific_name}</em>)</h2>
+                    </div>
+                </header>
+
+                <div className="panel">
+                    <dl className={itemClass}>
+                        {/* TODO link to NCBI Entrez page? */}
+                        <div data-test="gendid">
+                            <dt>Entrez GeneID</dt>
+                            <dd>{context.geneid}</dd>
+                        </div>
+
+                        {/* TODO link to NADB page? */}
+                        {context.symbol && geneLink ?
+                            <div data-test="symbol">
+                                <dt>Gene symbol</dt>
+                                <dd><a href={geneLink}>{context.symbol}</a></dd>
+                            </div>
+                        : null}
+
+                        {context.name ?
+                            <div data-test="name">
+                                <dt>Official gene name</dt>
+                                <dd>{context.name}</dd>
+                            </div>
+                        : null}
+
+                        {context.synonyms ?
+                          <div data-test="synonyms">
+                              <dt>Synonyms</dt>
+                              <dd><ul>
+                                  {context.synonyms.map((synonym, i) =>
```

Comment:
You can use `synonym` as the `<li>` key and remove `i` which would be preferable, though it’s not always possible to avoid using `i` as the key. If it’s possible two entries in the `synonyms` array could be the same, then you’d need to keep using `i`. The same thing applies to the `go_bp_ids` and `go_mf_id` arrays below.